### ISSUE TITLE
Update to Release Note Generation

### ISSUE THREAD

**Issue by halspang** [dapr/dapr#4044]
## Description
Recently, I made a small update to the Azure Go SDK. When I made this change I found their release note process to be interesting. Instead of tagging the issues with a release note tag, they instead had a change log. Updating the change log was a required part of the PR. My question is, would people be interested in trying this for Dapr? Or perhaps one of the SDK repos (I volunteer .NET as tribute).

The reason I feel this may be beneficial is we tend to rush the release notes at the very end of the release process. Though this does work, it seems like it might be a bit easier to do over the course of the release instead of all at once at the end.

This also gives the nice benefit of keeping track of what changes are in the upcoming release at a glance. 

Here's the SDK process:

[Change log](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azservicebus/CHANGELOG.md)
[Updating PR](https://github.com/Azure/azure-sdk-for-go/commit/7a0cef22a2ef73f165b199c7c636049aff677030)

### tl;dr
1. Have a changelog.md in the various repos.
2. Update as part of the PR process.
3. Aggregate the already reviewed release notes at release time.

**Comment by tanvigour** [comment]
I think this is a good suggestion. The current way of adding release notes at the end and sometimes that can get a bit tedious.

**Comment by berndverst** [comment]
This is a good idea.

We should have a GitHub Action in every repo that forces an update to the `changelog.md` with each PR.

As for release notes process itself:

The current release notes script would be updated to import the various `changelog.md` files. The code for determining contributors doesn't need to be changed though.

The process otherwise will be unchanged -- the generated release notes can be imported into HackMD, but at that point only release manager / captain and the PM team need to collaborate on writing the highlights of the release.

It will no longer be necessary to carefully review / edit the release notes here.

One small downside:
- What if an issue is being re-opened but no PR has been made? Who would make a PR to update the changelog to indicate a previous PR didn't actually resolve a particular issue?

**Comment by halspang** [comment]
>One small downside:
>What if an issue is being re-opened but no PR has been made? Who would make a PR to update the changelog to indicate a previous PR didn't actually resolve a particular issue?

So basically we're saying that the issue is re-opened because whatever was done for it didn't actually accomplish the issue's goal? I think the person re-opening the issue would be responsible. However, I don't see a ton of cases where this happens that it would invalidate the previous release notes where we could launch without either fixing it or reverting the offending commit.

**Comment by berndverst** [comment]
I'd be happy to try this in Contrib and Python SDK (as a maintainer for those). Step one would be to update the `Contributing.md` if present, and then to consistently enforce this with approvers and maintainers.

**Comment by halspang** [comment]
Contrib is an interesting one to offer as it has a lot of community traction. I can also do it for .NET :)

**Comment by halspang** [comment]
@berndverst - If we have them write the changelog manually, it could just include a link to their issue/PR. Should achieve the same thing, yeah?
